src/pages/guides/terraform/s3-encryption.mdx

```diff
@@ -0,0 +1,113 @@
+
+<Note>
+  Server-side encryption with Amazon S3 managed keys (SSE-S3) is free. Other
+  encryption options may incur costs. Amazon offers pricing details on their
+  [pricing page](https://aws.amazon.com/s3/pricing/).
+</Note>
+
+## What we'll be doing
+
+1. Review the existing module
+2. Configure server-side encryption (SSE)
+
+## Review the existing module
+
+Start by cloning the [Nitric repository](https://github.com/nitrictech/nitric), then examine how the Terrraform provider [provisions an S3 bucket](https://github.com/nitrictech/nitric/tree/main/cloud/aws/deploytf/.nitric/modules/bucket).
+
+```bash
+git clone https://github.com/nitrictech/nitric
+cd nitric
+```
+
+The AWS S3 module in the default Terraform provider performs the following tasks:
+
+1. Creates a unique ID for the S3 bucket to ensure unique naming.
+2. Provisions an S3 bucket with a unique name using the generated ID.
+3. Tags the bucket for identification.
+4. Grants S3 permission to invoke specified Lambda functions.
+5. Configures S3 bucket notifications to trigger Lambda functions based on specified events using dynamic blocks.
+
+To begin our customization, we will start adding configuration to this module.
+
+## Add server-side encryption (SSE)
+
+Add the following at the end of `aws/deploytf/.nitric/modules/bucket/main.tf`:
+
+```hcl {{ tag: 'aws/deploytf/.nitric/modules/bucket/main.tf' }}
+# AWS S3 bucket server-side encryption configuration
+resource "aws_s3_bucket_server_side_encryption_configuration" "bucket_encryption" {
+  bucket = aws_s3_bucket.bucket.bucket
+
+  rule {
+    apply_server_side_encryption_by_default {
+      sse_algorithm = "AES256"
+    }
+  }
+}
+```
+
+<Note>
+  Full documentation can be found on the [Terraform
+  registry](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_server_side_encryption_configuration).
+</Note>
+
+## Build and use your modified provider
+
+The Nitric project includes a make file that will build and install your provider as `nitric/awstf@0.0.1` by default.
+
+Navigate to `nitric/cloud/aws` and run `make install` to build and install the modified provider binary.
+
+```bash
+cd nitric/cloud/aws
+
+make install
+```
+
+The provider can then be used directly in your project's stack file as follows.
+
+```yaml
+# The nitric provider to use
+provider: nitric/awstf@0.0.1
+
+# The target aws region to deploy to
+region: us-east-2
+```
+
+<Note>
+  If you don't have a stack file use `nitric stack new` to create one.
+</Note>
+
+Because the [Terraform providers](https://nitric.io/docs/reference/providers/terraform) are in preview, you'll also need to enable `beta-providers` in your Nitric project by adding the following to your project's nitric.yaml file:
```

Comment:
```suggestion
Because the [Terraform providers](/reference/providers/terraform) are in preview, you'll also need to enable `beta-providers` in your Nitric project by adding the following to your project's nitric.yaml file:
```